SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialPlayerSpawner.cpp

```diff
@@ -198,40 +198,58 @@ void USpatialPlayerSpawner::ReceivePlayerSpawnRequestOnServer(const Worker_Comma
+	// 
+	// This implementation depends on:
+	// 1) the load-balancing strategy having the same rules for PlayerStart Actors and Characters / Controllers /
 	// Player States or,
-	// 2) the load-balancing strategy can change the authoritative virtual worker ID for a PlayerStart Actor
-	// during the lifetime of a deployment.
-	if (NetDriver->LoadBalanceStrategy != nullptr)
+	// 2) the authoritative virtual worker ID for a PlayerStart Actor not changing during the lifetime of a deployment.
+	check (NetDriver->LoadBalanceStrategy != nullptr)
+
+	// We need to specifically extract the URL from the PlayerSpawn request for finding a PlayerStart.
+	const FURL Url = PlayerSpawner::ExtractUrlFromPlayerSpawnParams(SpawnPlayerRequest);
+
+	// Find a PlayerStart Actor on this server.
+	AActor* PlayerStartActor = NetDriver->GetWorld()->GetAuthGameMode()->FindPlayerStart(nullptr, Url.Portal);
+
+	// If the PlayerStart is authoritative locally, spawn the player locally.
+	if (PlayerStartActor != nullptr && NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*PlayerStartActor))
 	{
-		// We need to specifically extract the URL from the PlayerSpawn request for finding a PlayerStart.
-		const FURL Url = PlayerSpawner::ExtractUrlFromPlayerSpawnParams(SpawnPlayerRequest);
-		AActor* PlayerStartActor = NetDriver->GetWorld()->GetAuthGameMode()->FindPlayerStart(nullptr, Url.Portal);
+		UE_LOG(LogSpatialPlayerSpawner, Verbose, TEXT("Handling SpawnPlayerRequest request locally. Client worker ID: %s."), *ClientWorkerId);
+		PassSpawnRequestToNetDriver(SpawnPlayerRequest, PlayerStartActor);
+		return;
+	}
 
-		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*PlayerStartActor))
-		{
-			// If we fail to forward the spawn request, we default to the normal player spawning flow.
-			const bool bSuccessfullyForwardedRequest = ForwardSpawnRequestToStrategizedServer(SpawnPlayerRequest, PlayerStartActor, ClientWorkerId);
-			if (bSuccessfullyForwardedRequest)
-			{
-				return;
-			}
-		}
-		else
-		{
-			UE_LOG(LogSpatialPlayerSpawner, Verbose, TEXT("Handling SpawnPlayerRequest request locally. Client worker ID: %s."), *ClientWorkerId);
-			PassSpawnRequestToNetDriver(SpawnPlayerRequest, PlayerStartActor);
-			return;
-		}
+	VirtualWorkerId VirtualWorkerToForwardTo = SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+
+	// If we can't find a PlayerStart Actor, the PlayerSpawner authoritative worker may be part of a layer
+	// which has deleted all PlayerStart Actors and/or shouldn't be processing player spawning. In this case,
+	// we attempt to forward to the worker authoritative over the PlayerStart CDO, was pass a null object ref
+	// so that the forwarded worker knows to search for a PlayerStart.
+	if (PlayerStartActor == nullptr)
+	{
+		VirtualWorkerToForwardTo = NetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(*Cast<AActor>(APlayerStart::StaticClass()->GetDefaultObject()));
```

Comment:
Remind me again, why we use the CDO and not the GameMode?

Reply:
Happy to do gamemode

Reply:
This should also resolve the problem of having multiple types extended from APlayerStart with different worker associations